### DISCUSSION TITLE
Use C++20 ranges if a compiler supports it

### DISCUSSION THREAD

**Discussion by data-man** [contour-terminal/contour#290] [Development]
Research and benchmarks are needed. :smile:

**Comment by christianparpart** [comment]
just like fmt::format vs std::format then, I guess.

we're doing the same with std::filesystem already, too. (wrt. non-supporting Mac's)

**Reply by data-man** [reply]
> just like fmt::format vs std::format then, I guess.

`fmt::format` support console color & styles (although we are not using this yet in cli).
Some time ago I proposed to improve this in https://github.com/fmtlib/fmt/issues/2302 and today https://github.com/fmtlib/fmt/pull/2394 was merged.
But as I mentioned earlier, I'd prefer to use [strf](https://github.com/robhz786/strf).

**Reply by WSLUser** [reply]
> But as I mentioned earlier, I'd prefer to use strf.

Well that was written for C++ 11.

FMT is part of the C++20 standard. So everything you can do with FMT you can do with C++20 by default. Use of FMT makes it easier to transition to C++20 once ready.  Other projects such as GSL already adopted C++20, which actually forced Windows Terminal to become C++20 compliant in their use of span.
